processing/src/main/java/io/druid/segment/serde/DictionaryEncodedColumnSupplier.java

```diff
@@ -52,7 +53,7 @@ public DictionaryEncodedColumn get()
     return new SimpleDictionaryEncodedColumn(
         singleValuedColumn,
         multiValuedColumn,
-        lookupCacheSize > 0 ? dictionary.withCache(lookupCacheSize) : dictionary
+        new CachingIndexed<>(dictionary, lookupCacheSize)
```

Comment:
Fwiw, keeping the original logic that has the if statement outside seems nicer to me.